uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Date/BasicDatePeriod.php

```diff
@@ -47,6 +58,9 @@ public function __construct($start, $end) {
     } else {
       $end = new DateTime($end);
     }
+    $this->startDate = $start;
+    $this->endDate = clone $end;
```

Comment:
Tried this and it was affecting some other function that was expecting DateTime Objects. Since the startDate and endDate properties are private already, then these properties can't be modified.